src/Features/CSharp/Portable/UseExpressionBody/UseExpressionBodyCodeRefactoringProvider.cs

```diff
@@ -152,8 +185,85 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
+            var (helper, useExpressionBody) = s_helperToTitleMap[equivalenceKey];
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var declarationsToFix = GetDeclarationsToFix(fixAllSpans, root, helper, useExpressionBody, optionSet);
+            await FixDeclarationsAsync(document, editor, root, declarationsToFix, helper, useExpressionBody, cancellationToken).ConfigureAwait(false);
+            return;
+
+            // Local functions.
+            static IEnumerable<SyntaxNode> GetDeclarationsToFix(
+                ImmutableArray<TextSpan> fixAllSpans,
+                SyntaxNode root,
+                UseExpressionBodyHelper helper,
+                bool useExpressionBody,
+                OptionSet optionSet)
+            {
+                foreach (var span in fixAllSpans)
+                {
+                    var spanNode = root.FindNode(span);
+
+                    foreach (var node in spanNode.DescendantNodesAndSelf())
+                    {
+                        if (!helper.IsRelevantDeclarationNode(node) || !helper.SyntaxKinds.Contains(node.Kind()))
+                            continue;
+
+                        if (useExpressionBody && helper.CanOfferUseExpressionBody(optionSet, node, forAnalyzer: false))
+                        {
+                            yield return node;
+                        }
+                        else if (!useExpressionBody && helper.CanOfferUseBlockBody(optionSet, node, forAnalyzer: false, out _, out _))
+                        {
+                            yield return node;
+                        }
+                    }
+                }
+            }
+
+            static async Task FixDeclarationsAsync(
+                Document document,
+                SyntaxEditor editor,
+                SyntaxNode root,
+                IEnumerable<SyntaxNode> declarationsToFix,
+                UseExpressionBodyHelper helper,
+                bool useExpressionBody,
+                CancellationToken cancellationToken)
+            {
+                // Process all declaration nodes in reverse to handle nested declaration updates properly.
+                declarationsToFix = declarationsToFix.Reverse();
+
+                // Track all the declaration nodes to be fixed so we can get the latest declaration node in the current root during updates.
+                var currentRoot = root.TrackNodes(declarationsToFix);
+
+                foreach (var declaration in declarationsToFix)
+                {
+                    // Get the current document, root, semanticModel and declaration.
+                    document = document.WithSyntaxRoot(currentRoot);
+                    currentRoot = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Is this necessary?

Reply:
Yes, we eventually end up reaching [here](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/CSharpSyntaxTree.cs,330), which creates a new `ParsedSyntaxTree` with `cloneRoot: true`, leading to the tree being created with a cloned root [here](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/CSharpSyntaxTree.ParsedSyntaxTree.cs,51).